autograder/submitty_autograding_shipper.py

```diff
@@ -1299,18 +1311,29 @@ def try_short_circuit(config: dict, queue_file: str) -> bool:
         course_path, 'config', 'form', f'form_{queue_obj["gradeable"]}.json'
     )
 
-    with open(config_path) as fd:
-        config_obj = json.load(fd)
-
-    if not can_short_circuit(config_obj):
+    # Some of the config files may disappear if we're running this step and the gradeable is
+    # currently in the process of being re-built. In this case, we give up on trying to
+    # short-circuit and let the full autograder handle it. Best-case scenario, the config once
+    # again exists and we can grade it normally. Worst-case scenario, the config is still out of
+    # commission, but the grader is better-equipped to handle this scenario.
+    try:
+        with open(config_path) as fd:
+            config_obj = json.load(fd)
+        with open(gradeable_config_path) as fd:
+            gradeable_config_obj = json.load(fd)
+    except Exception as e:
```

Comment:
Can we narrow the types of exceptions we expect here? Broadly catching any and all exceptions is not good practice as it ends up meaning we're going to run the risk of catching stuff we don't mean to. For example, if you mistyped `json.load` or `open` or whatever, it'll throw a `NameError` or an `AttributeError` which are both caught by `Exception` meaning we'll just constantly hit this except block even though the code is broken in a way we do not expect.

Instead, we should catch on as narrow a scope as possible. For example, within this block, we may expect we'd hit `OSError` and `json.JSONDecodeError` where the former covers still a wide swathe of conditions (cannot open, file permissions, it's a directory instead of a file, etc.), but it's narrower than `Exception` and limits only to OS related stuff. At best, doing this within dedicated code paths just strikes as "we don't actually know the side-effects of our code, let's catch it all".

I realize there's a lot of these `except Exception` catches within the autograding code, but please let's not add yet another that is definitely easy to avoid in this particular case.

Reply:
Narrowed it down to `FileNotFoundError` as that's the particular error that's not particularly a sign that something's gone wrong but rather a sign that the submission was a victim of unfortunate timing (being graded in the middle of a gradeable rebuild)